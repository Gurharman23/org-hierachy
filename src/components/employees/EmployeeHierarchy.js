import React from "react";
import { getColor } from "../../utilities";
import EmployeeCard from "./EmployeeCard";

export default function EmployeeHierarchy(props) {
  const root = props.employees.length > 0 ? props.employees[0] : null;

  return root === null ? (
    <></>
  ) : (
    <div>
      <div
        key={root.id}
        className="align-items-center d-flex flex-column p-2  m-1 border"
      >
        <EmployeeCard employee={root} colorIndex={0} />
        <div className="d-flex mt-4">
          {renderChilds(
            root.id,
            props.employees,
            props.deleteEmployee,
            props.changeRoute,
            0
          )}
        </div>
      </div>
    </div>
  );
}

function renderChilds(
  parentId,
  employees,
  deleteEmployee,
  changeRoute,
  parentColorIndex
) {
  const colorIndex = getColor(parentColorIndex);
  return (
    <>
      {employees.map((employee) => {
        if (parentId === employee.parent)
          return (
            <div
              key={employee.id}
              className="align-items-center d-flex flex-column  m-1 p-2 border"
            >
              <EmployeeCard
                employee={employee}
                deleteEmployee={deleteEmployee}
                changeRoute={changeRoute}
                colorIndex={colorIndex}
              />
              <div className="d-flex mt-4">
                {renderChilds(
                  employee.id,
                  employees,
                  deleteEmployee,
                  changeRoute,
                  colorIndex
                )}
              </div>
            </div>
          );
        // eslint-disable-next-line array-callback-return
        return;
      })}
    </>
  );
}
