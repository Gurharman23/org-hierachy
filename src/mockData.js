export const employees = [
  { id: 1, name: "Stella Payne Diaz", title: "CEO" },
  { id: 2, name: "Luke Warm", title: "VP Marketing/Sales", parent: 1 },
  { id: 3, name: "Meg Meehan Hoffa", title: "Sales", parent: 2 },
  { id: 4, name: "Peggy Flaming", title: "VP Engineering", parent: 1 },
  { id: 5, name: "Saul Wellingood", title: "Manufacturing", parent: 4 },
  { id: 6, name: "Al Ligori", title: "Marketing", parent: 2 },
  { id: 7, name: "Dot Stubadd", title: "Sales Rep", parent: 3 },
  { id: 8, name: "Les Ismore", title: "Project Mgr", parent: 5 },
  { id: 9, name: "April Lynn Parris", title: "Events Mgr", parent: 6 },
  { id: 10, name: "Xavier Breath", title: "Engineering", parent: 4 },
  { id: 11, name: "Anita Hammer", title: "Process", parent: 5 },
  { id: 12, name: "Billy Aiken", title: "Software", parent: 10 },
  { id: 13, name: "Stan Wellback", title: "Testing", parent: 10 },
  { id: 14, name: "Marge Innovera", title: "Hardware", parent: 10 },
  { id: 15, name: "Evan Elpus", title: "Quality", parent: 5 },
  { id: 16, name: "Lotta B. Essen", title: "Sales Rep", parent: 3 },
  {
    name: "Gurharmanjit Singh",
    title: "Process",
    parent: 13,
    id: 18,
  },
];
