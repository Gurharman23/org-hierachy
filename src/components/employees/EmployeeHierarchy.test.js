import React from "react";
import renderer from "react-test-renderer";
import { employees } from "../../mockData.js";
import EmployeeHierarchy from "./EmployeeHierarchy.js";
it("To check the rendered tree Hierarchy", () => {
  const tree = renderer.create(
    <EmployeeHierarchy
      employees={employees}
      deleteEmployee={jest.fn()}
      changeRoute={jest.fn()}
    />
  );

  expect(tree).toMatchSnapshot();
});
