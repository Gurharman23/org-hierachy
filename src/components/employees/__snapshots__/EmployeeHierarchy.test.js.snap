// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To check the rendered tree Hierarchy 1`] = `
<div>
  <div
    className="align-items-center d-flex flex-column p-2  m-1 border"
  >
    <div
      className="card card-dimensions position-relative"
      style={
        Object {
          "borderColor": "blue",
          "borderWidth": "0.125rem",
        }
      }
    >
      <div
        className="card-body"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        <h5
          className="card-title text-truncate"
        >
          Stella Payne Diaz
        </h5>
        <h6
          className="card-subtitle mb-2 text-muted text-truncate"
        >
          Title: 
          CEO
        </h6>
        <p
          className="m-0"
        >
           ID: 
          1
           
        </p>
        <div
          className="position-absolute top-0 btn m-0 p-0"
          onClick={[Function]}
          style={
            Object {
              "right": "0px",
            }
          }
        >
           
          <svg
            className="bi bi-x"
            fill="currentColor"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="d-flex mt-4"
    >
      <div
        className="align-items-center d-flex flex-column  m-1 p-2 border"
      >
        <div
          className="card card-dimensions position-relative"
          style={
            Object {
              "borderColor": "red",
              "borderWidth": "0.125rem",
            }
          }
        >
          <div
            className="card-body"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <h5
              className="card-title text-truncate"
            >
              Luke Warm
            </h5>
            <h6
              className="card-subtitle mb-2 text-muted text-truncate"
            >
              Title: 
              VP Marketing/Sales
            </h6>
            <p
              className="m-0"
            >
               ID: 
              2
               
            </p>
            <div
              className="position-absolute top-0 btn m-0 p-0"
              onClick={[Function]}
              style={
                Object {
                  "right": "0px",
                }
              }
            >
               
              <svg
                className="bi bi-x"
                fill="currentColor"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="d-flex mt-4"
        >
          <div
            className="align-items-center d-flex flex-column  m-1 p-2 border"
          >
            <div
              className="card card-dimensions position-relative"
              style={
                Object {
                  "borderColor": "green",
                  "borderWidth": "0.125rem",
                }
              }
            >
              <div
                className="card-body"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
              >
                <h5
                  className="card-title text-truncate"
                >
                  Meg Meehan Hoffa
                </h5>
                <h6
                  className="card-subtitle mb-2 text-muted text-truncate"
                >
                  Title: 
                  Sales
                </h6>
                <p
                  className="m-0"
                >
                   ID: 
                  3
                   
                </p>
                <div
                  className="position-absolute top-0 btn m-0 p-0"
                  onClick={[Function]}
                  style={
                    Object {
                      "right": "0px",
                    }
                  }
                >
                   
                  <svg
                    className="bi bi-x"
                    fill="currentColor"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="d-flex mt-4"
            >
              <div
                className="align-items-center d-flex flex-column  m-1 p-2 border"
              >
                <div
                  className="card card-dimensions position-relative"
                  style={
                    Object {
                      "borderColor": "pink",
                      "borderWidth": "0.125rem",
                    }
                  }
                >
                  <div
                    className="card-body"
                    onClick={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                      }
                    }
                  >
                    <h5
                      className="card-title text-truncate"
                    >
                      Dot Stubadd
                    </h5>
                    <h6
                      className="card-subtitle mb-2 text-muted text-truncate"
                    >
                      Title: 
                      Sales Rep
                    </h6>
                    <p
                      className="m-0"
                    >
                       ID: 
                      7
                       
                    </p>
                    <div
                      className="position-absolute top-0 btn m-0 p-0"
                      onClick={[Function]}
                      style={
                        Object {
                          "right": "0px",
                        }
                      }
                    >
                       
                      <svg
                        className="bi bi-x"
                        fill="currentColor"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  className="d-flex mt-4"
                />
              </div>
              <div
                className="align-items-center d-flex flex-column  m-1 p-2 border"
              >
                <div
                  className="card card-dimensions position-relative"
                  style={
                    Object {
                      "borderColor": "pink",
                      "borderWidth": "0.125rem",
                    }
                  }
                >
                  <div
                    className="card-body"
                    onClick={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                      }
                    }
                  >
                    <h5
                      className="card-title text-truncate"
                    >
                      Lotta B. Essen
                    </h5>
                    <h6
                      className="card-subtitle mb-2 text-muted text-truncate"
                    >
                      Title: 
                      Sales Rep
                    </h6>
                    <p
                      className="m-0"
                    >
                       ID: 
                      16
                       
                    </p>
                    <div
                      className="position-absolute top-0 btn m-0 p-0"
                      onClick={[Function]}
                      style={
                        Object {
                          "right": "0px",
                        }
                      }
                    >
                       
                      <svg
                        className="bi bi-x"
                        fill="currentColor"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  className="d-flex mt-4"
                />
              </div>
            </div>
          </div>
          <div
            className="align-items-center d-flex flex-column  m-1 p-2 border"
          >
            <div
              className="card card-dimensions position-relative"
              style={
                Object {
                  "borderColor": "green",
                  "borderWidth": "0.125rem",
                }
              }
            >
              <div
                className="card-body"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
              >
                <h5
                  className="card-title text-truncate"
                >
                  Al Ligori
                </h5>
                <h6
                  className="card-subtitle mb-2 text-muted text-truncate"
                >
                  Title: 
                  Marketing
                </h6>
                <p
                  className="m-0"
                >
                   ID: 
                  6
                   
                </p>
                <div
                  className="position-absolute top-0 btn m-0 p-0"
                  onClick={[Function]}
                  style={
                    Object {
                      "right": "0px",
                    }
                  }
                >
                   
                  <svg
                    className="bi bi-x"
                    fill="currentColor"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="d-flex mt-4"
            >
              <div
                className="align-items-center d-flex flex-column  m-1 p-2 border"
              >
                <div
                  className="card card-dimensions position-relative"
                  style={
                    Object {
                      "borderColor": "pink",
                      "borderWidth": "0.125rem",
                    }
                  }
                >
                  <div
                    className="card-body"
                    onClick={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                      }
                    }
                  >
                    <h5
                      className="card-title text-truncate"
                    >
                      April Lynn Parris
                    </h5>
                    <h6
                      className="card-subtitle mb-2 text-muted text-truncate"
                    >
                      Title: 
                      Events Mgr
                    </h6>
                    <p
                      className="m-0"
                    >
                       ID: 
                      9
                       
                    </p>
                    <div
                      className="position-absolute top-0 btn m-0 p-0"
                      onClick={[Function]}
                      style={
                        Object {
                          "right": "0px",
                        }
                      }
                    >
                       
                      <svg
                        className="bi bi-x"
                        fill="currentColor"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  className="d-flex mt-4"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="align-items-center d-flex flex-column  m-1 p-2 border"
      >
        <div
          className="card card-dimensions position-relative"
          style={
            Object {
              "borderColor": "red",
              "borderWidth": "0.125rem",
            }
          }
        >
          <div
            className="card-body"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <h5
              className="card-title text-truncate"
            >
              Peggy Flaming
            </h5>
            <h6
              className="card-subtitle mb-2 text-muted text-truncate"
            >
              Title: 
              VP Engineering
            </h6>
            <p
              className="m-0"
            >
               ID: 
              4
               
            </p>
            <div
              className="position-absolute top-0 btn m-0 p-0"
              onClick={[Function]}
              style={
                Object {
                  "right": "0px",
                }
              }
            >
               
              <svg
                className="bi bi-x"
                fill="currentColor"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          className="d-flex mt-4"
        >
          <div
            className="align-items-center d-flex flex-column  m-1 p-2 border"
          >
            <div
              className="card card-dimensions position-relative"
              style={
                Object {
                  "borderColor": "green",
                  "borderWidth": "0.125rem",
                }
              }
            >
              <div
                className="card-body"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
              >
                <h5
                  className="card-title text-truncate"
                >
                  Saul Wellingood
                </h5>
                <h6
                  className="card-subtitle mb-2 text-muted text-truncate"
                >
                  Title: 
                  Manufacturing
                </h6>
                <p
                  className="m-0"
                >
                   ID: 
                  5
                   
                </p>
                <div
                  className="position-absolute top-0 btn m-0 p-0"
                  onClick={[Function]}
                  style={
                    Object {
                      "right": "0px",
                    }
                  }
                >
                   
                  <svg
                    className="bi bi-x"
                    fill="currentColor"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="d-flex mt-4"
            >
              <div
                className="align-items-center d-flex flex-column  m-1 p-2 border"
              >
                <div
                  className="card card-dimensions position-relative"
                  style={
                    Object {
                      "borderColor": "pink",
                      "borderWidth": "0.125rem",
                    }
                  }
                >
                  <div
                    className="card-body"
                    onClick={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                      }
                    }
                  >
                    <h5
                      className="card-title text-truncate"
                    >
                      Les Ismore
                    </h5>
                    <h6
                      className="card-subtitle mb-2 text-muted text-truncate"
                    >
                      Title: 
                      Project Mgr
                    </h6>
                    <p
                      className="m-0"
                    >
                       ID: 
                      8
                       
                    </p>
                    <div
                      className="position-absolute top-0 btn m-0 p-0"
                      onClick={[Function]}
                      style={
                        Object {
                          "right": "0px",
                        }
                      }
                    >
                       
                      <svg
                        className="bi bi-x"
                        fill="currentColor"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  className="d-flex mt-4"
                />
              </div>
              <div
                className="align-items-center d-flex flex-column  m-1 p-2 border"
              >
                <div
                  className="card card-dimensions position-relative"
                  style={
                    Object {
                      "borderColor": "pink",
                      "borderWidth": "0.125rem",
                    }
                  }
                >
                  <div
                    className="card-body"
                    onClick={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                      }
                    }
                  >
                    <h5
                      className="card-title text-truncate"
                    >
                      Anita Hammer
                    </h5>
                    <h6
                      className="card-subtitle mb-2 text-muted text-truncate"
                    >
                      Title: 
                      Process
                    </h6>
                    <p
                      className="m-0"
                    >
                       ID: 
                      11
                       
                    </p>
                    <div
                      className="position-absolute top-0 btn m-0 p-0"
                      onClick={[Function]}
                      style={
                        Object {
                          "right": "0px",
                        }
                      }
                    >
                       
                      <svg
                        className="bi bi-x"
                        fill="currentColor"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  className="d-flex mt-4"
                />
              </div>
              <div
                className="align-items-center d-flex flex-column  m-1 p-2 border"
              >
                <div
                  className="card card-dimensions position-relative"
                  style={
                    Object {
                      "borderColor": "pink",
                      "borderWidth": "0.125rem",
                    }
                  }
                >
                  <div
                    className="card-body"
                    onClick={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                      }
                    }
                  >
                    <h5
                      className="card-title text-truncate"
                    >
                      Evan Elpus
                    </h5>
                    <h6
                      className="card-subtitle mb-2 text-muted text-truncate"
                    >
                      Title: 
                      Quality
                    </h6>
                    <p
                      className="m-0"
                    >
                       ID: 
                      15
                       
                    </p>
                    <div
                      className="position-absolute top-0 btn m-0 p-0"
                      onClick={[Function]}
                      style={
                        Object {
                          "right": "0px",
                        }
                      }
                    >
                       
                      <svg
                        className="bi bi-x"
                        fill="currentColor"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  className="d-flex mt-4"
                />
              </div>
            </div>
          </div>
          <div
            className="align-items-center d-flex flex-column  m-1 p-2 border"
          >
            <div
              className="card card-dimensions position-relative"
              style={
                Object {
                  "borderColor": "green",
                  "borderWidth": "0.125rem",
                }
              }
            >
              <div
                className="card-body"
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
              >
                <h5
                  className="card-title text-truncate"
                >
                  Xavier Breath
                </h5>
                <h6
                  className="card-subtitle mb-2 text-muted text-truncate"
                >
                  Title: 
                  Engineering
                </h6>
                <p
                  className="m-0"
                >
                   ID: 
                  10
                   
                </p>
                <div
                  className="position-absolute top-0 btn m-0 p-0"
                  onClick={[Function]}
                  style={
                    Object {
                      "right": "0px",
                    }
                  }
                >
                   
                  <svg
                    className="bi bi-x"
                    fill="currentColor"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              className="d-flex mt-4"
            >
              <div
                className="align-items-center d-flex flex-column  m-1 p-2 border"
              >
                <div
                  className="card card-dimensions position-relative"
                  style={
                    Object {
                      "borderColor": "pink",
                      "borderWidth": "0.125rem",
                    }
                  }
                >
                  <div
                    className="card-body"
                    onClick={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                      }
                    }
                  >
                    <h5
                      className="card-title text-truncate"
                    >
                      Billy Aiken
                    </h5>
                    <h6
                      className="card-subtitle mb-2 text-muted text-truncate"
                    >
                      Title: 
                      Software
                    </h6>
                    <p
                      className="m-0"
                    >
                       ID: 
                      12
                       
                    </p>
                    <div
                      className="position-absolute top-0 btn m-0 p-0"
                      onClick={[Function]}
                      style={
                        Object {
                          "right": "0px",
                        }
                      }
                    >
                       
                      <svg
                        className="bi bi-x"
                        fill="currentColor"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  className="d-flex mt-4"
                />
              </div>
              <div
                className="align-items-center d-flex flex-column  m-1 p-2 border"
              >
                <div
                  className="card card-dimensions position-relative"
                  style={
                    Object {
                      "borderColor": "pink",
                      "borderWidth": "0.125rem",
                    }
                  }
                >
                  <div
                    className="card-body"
                    onClick={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                      }
                    }
                  >
                    <h5
                      className="card-title text-truncate"
                    >
                      Stan Wellback
                    </h5>
                    <h6
                      className="card-subtitle mb-2 text-muted text-truncate"
                    >
                      Title: 
                      Testing
                    </h6>
                    <p
                      className="m-0"
                    >
                       ID: 
                      13
                       
                    </p>
                    <div
                      className="position-absolute top-0 btn m-0 p-0"
                      onClick={[Function]}
                      style={
                        Object {
                          "right": "0px",
                        }
                      }
                    >
                       
                      <svg
                        className="bi bi-x"
                        fill="currentColor"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  className="d-flex mt-4"
                >
                  <div
                    className="align-items-center d-flex flex-column  m-1 p-2 border"
                  >
                    <div
                      className="card card-dimensions position-relative"
                      style={
                        Object {
                          "borderColor": "yellow",
                          "borderWidth": "0.125rem",
                        }
                      }
                    >
                      <div
                        className="card-body"
                        onClick={[Function]}
                        style={
                          Object {
                            "cursor": "pointer",
                          }
                        }
                      >
                        <h5
                          className="card-title text-truncate"
                        >
                          Gurharmanjit Singh
                        </h5>
                        <h6
                          className="card-subtitle mb-2 text-muted text-truncate"
                        >
                          Title: 
                          Process
                        </h6>
                        <p
                          className="m-0"
                        >
                           ID: 
                          18
                           
                        </p>
                        <div
                          className="position-absolute top-0 btn m-0 p-0"
                          onClick={[Function]}
                          style={
                            Object {
                              "right": "0px",
                            }
                          }
                        >
                           
                          <svg
                            className="bi bi-x"
                            fill="currentColor"
                            height="16"
                            viewBox="0 0 16 16"
                            width="16"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                    <div
                      className="d-flex mt-4"
                    />
                  </div>
                </div>
              </div>
              <div
                className="align-items-center d-flex flex-column  m-1 p-2 border"
              >
                <div
                  className="card card-dimensions position-relative"
                  style={
                    Object {
                      "borderColor": "pink",
                      "borderWidth": "0.125rem",
                    }
                  }
                >
                  <div
                    className="card-body"
                    onClick={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                      }
                    }
                  >
                    <h5
                      className="card-title text-truncate"
                    >
                      Marge Innovera
                    </h5>
                    <h6
                      className="card-subtitle mb-2 text-muted text-truncate"
                    >
                      Title: 
                      Hardware
                    </h6>
                    <p
                      className="m-0"
                    >
                       ID: 
                      14
                       
                    </p>
                    <div
                      className="position-absolute top-0 btn m-0 p-0"
                      onClick={[Function]}
                      style={
                        Object {
                          "right": "0px",
                        }
                      }
                    >
                       
                      <svg
                        className="bi bi-x"
                        fill="currentColor"
                        height="16"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <div
                  className="d-flex mt-4"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
